doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+
+fpmsyncd and fdbsyncd also need to wait for some time for the BGP protocol to converge and populate the routing info from peers, before they can reconcile their respective app_db entries. Hence they start a fixed 120 seconds timer once BGP is started. Once this timer expires and orchagent is reconciled, they start the app_db reconciliation process.
+
+The below detailed sequence diagram explains the interaction during the restore, replay and reconcile phase.
+
+![System Warm reboot and reconciliation sequence](images/warmrebootseq2.PNG "Figure 17: System Warm reboot and reconciliation sequence detail")
+
+__Figure 17: Warm reboot replay sequence__
+
+To achieve this dependency new states "replayed", "reconciled"  and "disabled" are added to the WARM_RESTART_TABLE  table in STATE_DB.  The value "disabled" is set when the subsystems perform a cold restart.
+
+Producer: multiple producers
+
+Consumer: multiple consumers
+
+Description:
+
+- Value "replayed" "reconciled"  and "disabled" are added to the existing "state" variable.
+
+Schema:
+
+```
+; Existing table
+key             = WARM_RESTART_TABLE|<application_name>   
+; field         = value
+restore_count   = number    ;uint32
+state           = "restored"/"replayed"/"reconciled"/"disabled" 
+```
+
+**Configuration for EVPN  Warm reboot** 
+
+To support EVPN Warm reboot, system warm reboot must be enabled and BGP-GR must be configured for BGP on all the BGP peer nodes.
+
+The BGP GR restart and stale-path timers have a default value of 240 seconds and 720 seconds respectively.  To scale the system higher, these values may need to be tuned .  The restart timer should be long enough for the system to warm reboot ,  the IP connectivity to be re-established in the control plane and the data replay of the subsystems to the kernel to be complete.
+
+```
+BGP GR configuration for FRR
+router bgp <AS-NUM>
+ bgp graceful-restart
+ bgp graceful-restart preserve-fw-state
+ bgp graceful-restart stalepath-time 600  ( default 720)
+ bgp graceful-restart restart-time 300    ( default 240)
+```
+
+
+
+2. BGP Docker Warm reboot for EVPN
+   The goal of the BGP Docker warm reboot is to achieve BGP docker restart without impacting the dataplane. This is achieved with the help of graceful restart of the control plane (BGP).  As soon as BGP docker restarts, zebra re-learns the EVPN entries from kernel and marks them as stale. BGP protocol then re-learns the entries from the BGP peers and marks them for reconcile in zebra.  At the end of retention timer zebra  sweeps stale entries from the kernel. The remaining reconciliation sequence is same as the system warm-reboot, except that in this case only BGP docker and related tables are involved and data is already available in the DB and kernel. 
```

Comment:
Why " zebra re-learns the EVPN entries from kernel"?  appDB has all the routes, no need for Zebra to read them from kernel.

Reply:
In normal processing, zebra learns the Local EVPN routes from kernel and gives the info to BGP, which then sends these EVPN routes to the BGP EVPN peers. After wam-reboot, the local routes in BGP/Zebra are lost and hence zebra needs to learn them again from kernel so that it can give that info to BGP. BGP will then re-send these to its EVPN peers so that the BGP-GR process can successfully happen for its peers.

Reply:
So this "relearn from Zebra" is for local routes only? not including routes advertised by its peers?

Reply:
WG discussion: 
"-R" retains the routes but if this option is not present, kernel cleans up the routes when zebra dies. Suggest to add an option for retaining vs cleaning. If its without "-R" option. This would require fdbsyncd to be aware of bgp docker warmrestart. In this release, we can go with the default without "-R" option

Reply:
WG discussion: (10/23):
1. If there are no l2, then there is no additional wait time. The wait is only for the local address to be populated. 
2. Wait time:  There is a fixed wait time of 240sec, @nikhil can you confirm on this? Default BGP time is 120sec
3. User to ensure bgp graceful restart time and neighbor recovery timing are aligned

Reply:
@prsunny   Please find my reply below
1. Yes, no additional wait time when no L2 entries.
2. There is no fixed wait time. Just for fdbsyncd the reconcile time is 120 seconds and yes for BGP GR the default reconnect time is set to 120 seconds ( basically not change there from existing behaviour)
3. Yes, for EVPN config, to account for the dependency wait, GR timers need to be increased accordingly